Add Algeria holidays

abh31000 opened the pull request:
<!--
  Thanks for contributing to python-holidays!
-->

## Proposed change

<!--
  Describe the big picture of your changes.
  Don't forget to link your PR to an existing issue if any.
-->

Added Algeria holidays
Closes #1145 
Stuck at 82% coverage. Can anyone tell me what I'm missing here?

## Type of change

<!--
  Type of change you want to introduce. Please, check one (1) box only!
  If your PR requires multiple boxes to be checked, most likely it needs to
  be split into multiple PRs.
-->

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

<!--
  Put an `x` in the boxes that apply. You can change them after PR is created.
-->

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

<!--
  Thanks again for your contribution!
-->

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

KJhellico commented:
@abh31000, thank you for your contribution!

Seems your code based on `master` branch. You need to merge latest `beta` and add new line to `holidays/registry.py` for Algeria.

KJhellico commented on holidays/countries/algeria.py:
```diff
@@ -0,0 +1,85 @@
+        # Labour Day
+        self._add_labor_day("Labour Day")
+
+        # Independance Day
+        if year >= 1962:
+            self._add_holiday("Independance Day", JUL, 5)
```
```suggestion
        # Independence Day
        if year >= 1962:
            self._add_holiday("Independence Day", JUL, 5)
```

KJhellico commented on holidays/countries/algeria.py:
```diff
@@ -0,0 +1,85 @@
+
+        # As of April 30, 2023. Algeria has 3 days of Eid holidays
+        # (https://www.horizons.dz/english/archives/amp/12021)
+        if year < 2023:
+            # Eid al-Fitr - Feast Festive
+            self._add_eid_al_fitr_day("Eid al-Fitr")
+            self._add_eid_al_fitr_day_two("Eid al-Fitr Holiday")
+            # Eid al-Adha - Scarfice Festive
+            self._add_eid_al_adha_day("Eid al-Adha")
+            self._add_eid_al_adha_day_two("Eid al-Adha Holiday")
+
+        elif year == 2023:
+            # Eid al-Fitr - Feast Festive
+            self._add_eid_al_fitr_day("Eid al-Fitr")
+            self._add_eid_al_fitr_day_two("Eid al-Fitr Holiday")
+            # Eid al-Adha - Scarfice Festive
+            self._add_eid_al_adha_day("Eid al-Adha")
+            self._add_eid_al_adha_day_two("Eid al-Adha Holiday")
+            self._add_eid_al_adha_day_three("Eid al-Adha Holiday")
+
+        elif year > 2023:
+            # Eid al-Fitr - Feast Festive
+            self._add_eid_al_fitr_day("Eid al-Fitr")
+            self._add_eid_al_fitr_day_two("Eid al-Fitr Holiday")
+            self._add_eid_al_fitr_day_three("Eid al-Fitr Holiday")
+            # Eid al-Adha - Scarfice Festive
+            self._add_eid_al_adha_day("Eid al-Adha")
+            self._add_eid_al_adha_day_two("Eid al-Adha Holiday")
+            self._add_eid_al_adha_day_three("Eid al-Adha Holiday")
```
```suggestion
        # Eid al-Fitr - Feast Festive
        self._add_eid_al_fitr_day("Eid al-Fitr")
        self._add_eid_al_fitr_day_two("Eid al-Fitr Holiday")
        if year >= 2024:
            self._add_eid_al_fitr_day_three("Eid al-Fitr Holiday")

        # Eid al-Adha - Scarfice Festive
        self._add_eid_al_adha_day("Eid al-Adha")
        self._add_eid_al_adha_day_two("Eid al-Adha Holiday")
        if year >= 2023:
            self._add_eid_al_adha_day_three("Eid al-Adha Holiday")
```

KJhellico commented on holidays/countries/algeria.py:
```diff
@@ -0,0 +1,85 @@
+            self._add_holiday("Independance Day", JUL, 5)
+
+        # Revolution Day
+        if year > 1962:
```
```suggestion
        if year >= 1963:
```

KJhellico commented on holidays/countries/algeria.py:
```diff
@@ -0,0 +1,70 @@
+            self._add_holiday("Independence Day", JUL, 5)
+
+        # Revolution Day
+        if year > 1962:
```
```suggestion
        if year >= 1963:
```

arkid15r commented on tests/countries/test_algeria.py:
```diff
@@ -0,0 +1,72 @@
+        self.assertCountryAliases(Algeria, DZ, DZA)
+
+    def test_2022(self):
+        self.assertHoliday("2022-01-01")
+        self.assertHoliday("2022-01-12")
+        self.assertHoliday("2022-05-01")
+        self.assertHoliday("2022-05-02")
+        self.assertHoliday("2022-05-03")
+        self.assertHoliday("2022-07-05")
+        self.assertHoliday("2022-07-09")
+        self.assertHoliday("2022-07-10")
+        self.assertHoliday("2022-11-01")
```
The TestCase:assertHoliday was designed to accept multiple date-like strings:
```suggestion
        self.assertHoliday(
            "2022-01-01",
            "2022-01-12",
            "2022-05-01",
            "2022-05-02",
            "2022-05-03",
            "2022-07-05",
            "2022-07-09",
            "2022-07-10",
            "2022-11-01",
        )
```

Plese group them where possible.

arkid15r commented on holidays/countries/algeria.py:
```diff
@@ -0,0 +1,70 @@
+from holidays.holiday_groups import IslamicHolidays, InternationalHolidays
+
+
+class Algeria(HolidayBase, IslamicHolidays, InternationalHolidays):
```
```suggestion
class Algeria(HolidayBase, InternationalHolidays, IslamicHolidays):
```

arkid15r reviewed (commented):
Hi @abh31000,
thank you so much for implementing the Algeria holidays!

We're excited to see a new contributor! Also thanks a lot for addressing all the comments in such a timely manner.

I left a couple of minor comments below. Otherwise your code looks good!

P.S. as for the coverage issue you had with your first commit of this PR -- you can find out the  uncovered lines by running `make coverage` (which is an alias for `pytest --cov=. --cov-config=pyproject.toml --cov-report term-missing --dist loadscope --no-cov-on-fail --numprocesses auto`)

abh31000 commented:
> Hi @abh31000, thank you so much for implementing the Algeria holidays!
> 
> We're excited to see a new contributor! Also thanks a lot for addressing all the comments in such a timely manner.
> 
> I left a couple of minor comments below. Otherwise your code looks good!
> 
> P.S. as for the coverage issue you had with your first commit of this PR -- you can find out the uncovered lines by running `make coverage` (which is an alias for `pytest --cov=. --cov-config=pyproject.toml --cov-report term-missing --dist loadscope --no-cov-on-fail --numprocesses auto`)

You're welcome!
And now with this commit the test coverage is 100%.

arkid15r reviewed (commented):
🇩🇿 LGTM
